File: python/speedtest.py
import time

# The average execution time of this script is 3928.1 ms after 10 test

ITERATIONS: int = 10
NUMBER: int = 100_000_000

def speedtest() -> float:
	start: float = time.time() * 1000

	count = 0
	for _ in range (NUMBER):
		count += 1

	duration: float = round(time.time() * 1000 - start)

	return duration


def main():
	averageExecutionTime: float = 0
	finalDuration: float = 0

	print("Starting speed test...")

	for i in range(ITERATIONS):
		finalDuration = speedtest()
		averageExecutionTime += finalDuration
		print(f"{i + 1}. Counting to {NUMBER} in Python took {finalDuration} ms")
	
	averageExecutionTime /= ITERATIONS

	print(f"\n\033[32mAverage execution time: {averageExecutionTime} ms.")


if __name__ == "__main__":
	main()

File: c/speedtest.c
#include <stdio.h>
#include <time.h>

// The average execution time of this script is 368.79 ms after 10 test

const int NUMBER = 100000000;
const int ITERATIONS = 10;

float speedtest() {
    struct timespec start, end;

	clock_gettime(CLOCK_REALTIME, &start);

    int count = 0;
    for (int i = 0; i < NUMBER; i++) {
        count++;
    }

    clock_gettime(CLOCK_REALTIME, &end);

    long long start_ns = start.tv_sec * 1e9 + start.tv_nsec;
    long long end_ns = end.tv_sec * 1e9 + end.tv_nsec;

    return (end_ns - start_ns) / 1e6;
}

int main() {
	float averageExecutionTime = 0, finalDuration = 0;

	printf("Starting speed test...\n");

	for(int i = 0; i < ITERATIONS; i++){
		finalDuration = speedtest();
		averageExecutionTime += finalDuration;
		printf("%d. Counting to number %d in C took %.2f ms\n", i + 1, NUMBER, finalDuration);
	}

	averageExecutionTime /= ITERATIONS;

	printf("\n\e[0;32mAverage execution time: %.2f ms.\e[0m\n", averageExecutionTime);

	return 0;
}
